src/util/worker/mod.rs

```diff
@@ -275,22 +283,82 @@ where
     runner.shutdown();
 }
 
-// Fill buffer with next task batch comes from `rx`.
-fn fill_task_batch<T>(rx: &Receiver<Option<T>>, buffer: &mut Vec<T>, batch_size: usize) -> bool {
-    match rx.recv() {
-        Ok(Some(task)) => {
-            buffer.push(task);
-            while buffer.len() < batch_size {
-                match rx.try_recv() {
-                    Ok(Some(t)) => buffer.push(t),
-                    Err(TryRecvError::Empty) => break,
-                    Ok(None) | Err(_) => return false,
+fn poll_with_timer<R, T, U>(
```

Comment:
duplicated code with poll function, can we combine them? E.g, for poll, we can construct an empty timer